docs/examples/howto/add_new_algorithm.rst

```diff
@@ -0,0 +1,174 @@
+
+
+..  tip:: It is a common practice to put the algorithm code in ``ActsExamples`` namespace.
+
+Input and Output
+------------------
+
+The algorithm would be typically part of some processing sequence and thus consume and produce some event data.
+In hypothetical example discussed here space-points cod be the input and track seeds would be an output.
+The data can be retrieved in the algorithm using `get` method.
+
+.. code-block:: cpp
+
+    template<typename T>
+    const T& get(const std::string& name);
+
+The data is fetched from so called "store" that is populated by preceding algorithm or by the reader.
+
+The data object (or objects) produced by an algorithm can be placed in the store using method `add`.
+
+.. code-block:: cpp
+
+    template<typename T>
+    void add(const std::string& name, T&& object);
+
+As the method signature suggests the ownership of the object is transferred to the store.
+That is, the destruction of this object at the end of event processing is taken care of.
+
+Configurability
+----------------
+
+It is customary that an algorithm requires configuration parameters.
+For the sake of an example in hypothetical seeding algorithm it could be parameter steering
+which detector layers should be used. 
+The configuration can be provided to an algorithm through an additional class/structure.
+It can be an inner class of the algorithm or it can be external to it.
+One would use definitely external structure if it is shared among several algorithm classes.
+
+For example that is how the configuration object could look like for `MySeedingAlgorithm`:
+
+.. code-block:: cpp
+
+    struct MySeedingConfig {
+        std::vector<int> layers; // layers to use by the seeder
+        float deltaZ;  // the maximum allowed deviation in r-z plane
+    };
+
+.. tip:: It is customary to put the config structures in ``Acts`` namespace.
+
+The algorithm constructor would take `MySeedingConfig` object during the construction in addition 
+to the argument controlling verbosity of diagnostic messages.
+
+.. code-block:: cpp
+
+    MySeedingAlgorithm::MySeedingAlgorithm( Acts::MySeedingConfig cfg, Acts::Logging::Level lvl):
+      ActsExamples::BareAlgortihm("MySeedingAlgorithm", lvl), 
+      m_cfg(std::move(cfg)){...}
+
+
+Python bindings
+---------------
+In order to use an algorithm in standalone ACTS 
+the algorithm and the associated config structure need to be accessible from python.
+For that so called python bindings need to be created using ``pybind11`` library.
+The binding is defined in C++ code in `Examples/Python/src/` directory. 
+There is one source file per category, in this particular case the file to edit would be `TrackFinding.cpp`.
+
+The configuration structure binding would be defined like this:
+
+.. code-block:: cpp
+
+    using Config = Acts::MySeedingConfig;
+    auto c = py::class_<Config>(m, "MySeedingConfig").def(py::init<>()); // this defined the name know in python
```

Comment:
I'm not sure what the comment is supposed to be here, maybe "this definition is now known in python"? or something else?